app/controllers/api/base_controller/parser.rb

```diff
@@ -39,7 +39,7 @@ def validate_api_action
       end
 
       def parse_id(resource, collection)
-        return nil if resource.blank?
+        return nil if !resource.kind_of?(Hash) || resource.blank?
```

Comment:
You don't need `resource.blank?` anymore, though it's fine that it short circuits the `case` below.